lib/ansible/plugins/inventory/azure_rm.py

```diff
@@ -580,10 +579,11 @@ def __init__(self, nic_model, inventory_client, is_primary=False):
 
         self.public_ips = {}
 
-        for ipc in nic_model['properties']['ipConfigurations']:
-            pip = ipc['properties'].get('publicIPAddress')
-            if pip:
-                self._inventory_client._enqueue_get(url=pip['id'], api_version=self._inventory_client._network_api_version, handler=self._on_pip_response)
+        if nic_model.get('properties') and nic_model.get('properties').get('ipConfigurations'):
```

Comment:
maybe should be get('properties', {}).get('ipConfigurations')
otherwise first get may return None and it will crash anyway